Make json array generation more flexible

posionus opened the pull request:
Makes the regex generated for json arrays more valid and flexible.

1. Allows for empty lists when min_items == 0
2. Allows whitespace between array elements
3. Adds support for maxItems (before, having any maxItems != minItems would allow infinite repeats)

rlouf commented:
Thank you for opening a PR! Would you mind adding unit tests for the features/behaviors you implemented?

posionus commented:
> Thank you for opening a PR! Would you mind adding unit tests for the features/behaviors you implemented?

Added a new test and fixed existing tests

rlouf commented:
Thank you again for your contribution!
